app/Helper/FormHelper.php

```diff
@@ -232,7 +232,7 @@ public function textEditor($name, $values = array(), array $errors = array(), ar
         }
 
         $html = '<div class="js-text-editor" data-params=\''.json_encode($params, JSON_HEX_APOS).'\'>';
-        $html .= '<script type="text/template">'.(isset($values[$name]) ? $values[$name] : '').'</script>';
+        $html .= '<script type="text/template">'.(isset($values[$name]) ? htmlspecialchars($values[$name], ENT_QUOTES, 'UTF-8', true) : '').'</script>';
```

Comment:
I used htmlspecialchars instead of `$this->helper->text->e()` because the `e()` function calls `htmlspecialchars` with `$double_encode = FALSE` and it causes a wrong behaviour when the text contains an already escaped html, like `&amp;`. If you dont double encode id, the editor will load the escaped character `&` instead of the original text.
This only affects the edition, the visualization is still the same: If you type `&gt; &amp;` to a markdown text editor, you will see `> &` on task and comments view.
I tought of changing the `function e`, adding an optional parameter `$double_encode = true`.
If necessary, I'will change the textHelper and use it here.